contrib/codegen-tools/codegen/src/main/ops/org/nd4j/codegen/mixins/Mixins.kt

```diff
@@ -94,31 +94,67 @@ val scalar = Mixin("scalar"){
 val reduce = Mixin("reduce"){
     legacy = true
     Input(DataType.NUMERIC, "in") { description = "Input variable" }
+    Arg(DataType.BOOL,"keepDims"){"Whether to keep the original  dimensions or produce a shrunk array with less dimensions"}
     Arg(DataType.INT, "dimensions"){ count = AtLeast(0); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
     Output(DataType.NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
 }
 
+val reduceVariableDimensions = Mixin("reduceVariable") {
+    legacy = true
+    Input(DataType.NUMERIC, "in") { description = "Input variable" }
+    Input(DataType.NUMERIC, name = "dimensions"){description = "Dimensions to reduce along"; defaultValue = null }
+    Arg(DataType.BOOL,"keepDims"){"Whether to keep the original  dimensions or produce a shrunk array with less dimensions"}
+    Output(DataType.NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
+}
+
 val reduceFloating = Mixin("reduceFloating"){
     useMixin(reduce)
-    javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.floating"
+    javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.floating.custom"
```

Comment:
This will probably break some backwards compatibility - at least for people that are using Op objects directly.

I think we can probably accept that breakage, as most people will be using the factory methods and for them it should be transparent.